client/post-editor/editor-confirmation-sidebar/index.jsx

```diff
@@ -3,54 +3,113 @@
  */
 import React from 'react';
 import classnames from 'classnames';
+import { connect } from 'react-redux';
 import { localize } from 'i18n-calypso';
 
 /**
  * Internal dependencies
  */
 import RootChild from 'components/root-child';
 import Button from 'components/button';
+import EditorVisibility from 'post-editor/editor-visibility';
+import { editPost } from 'state/posts/actions';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { getEditorPostId } from 'state/ui/editor/selectors';
+import { getEditedPost } from 'state/posts/selectors';
 
 class EditorConfirmationSidebar extends React.Component {
 	static propTypes = {
-		hideSidebar: React.PropTypes.func,
-		isActive: React.PropTypes.bool,
+		closeOverlay: React.PropTypes.func,
+		closeSidebar: React.PropTypes.func,
+		isOverlayActive: React.PropTypes.bool,
+		isSidebarActive: React.PropTypes.bool,
 		onPublish: React.PropTypes.func,
+		post: React.PropTypes.object,
+		site: React.PropTypes.object,
+		savedPost: React.PropTypes.object,
+		onPrivatePublish: React.PropTypes.func,
+	};
+
+	closeOverlayAndSidebar = () => {
+		this.props.closeSidebar();
+		this.props.closeOverlay();
 	};
 
 	closeAndPublish = () => {
-		this.props.hideSidebar();
+		this.closeOverlayAndSidebar();
 		this.props.onPublish( true );
 	};
 
+	renderPrivacyControl() {
+		if ( ! this.props.post ) {
+			return;
+		}
+
+		const { password, type, status } = this.props.post || {};
+		const isPrivateSite = this.props.site && this.props.site.is_private;
```

Comment:
Could do `get( this.props, 'site.is_private', false )` -- I consider that slightly more readable, but it's largely a matter of personal preference.